src/ui.data-table/storybook/stories.js

```diff
@@ -206,6 +206,75 @@ Default.args = {};
 export const Nesting = DefaultTemplate.bind({});
 Nesting.args = { row: getNestedRow, selectable: true };
 
+export const NestedContent = DefaultTemplate.bind({});
+NestedContent.args = {
+  columns: [
+    { key: "key_1", label: "title 1" },
+    { key: "key_2", label: "title 2" },
+    { key: "key_3", label: "title 3" },
+  ],
+  row: (index) => {
+    if (index === 0) {
+      return {
+        id: getRandomId(),
+        isChecked: false,
+        key_1: {
+          primary: "Row with nested content",
+          secondary: "Click to expand",
+        },
+        key_2: {
+          primary: "primary",
+          secondary: "secondary",
+        },
+        key_3: {
+          primary: "primary",
+          secondary: "secondary",
+        },
+        nestedData: {
+          isChecked: false,
+          isHidden: true,
+          key_1: "Custom nesting #1",
+          key_2: "Custom nesting #2",
+          key_3: "Custom nesting #3",
+        },
+      };
+    } else {
+      return {
+        id: getRandomId(),
+        isChecked: false,
+        key_1: {
+          primary: `Regular row ${index}`,
+          secondary: "No nested content",
+        },
+        key_2: {
+          primary: "primary",
+          secondary: "secondary",
+        },
+        key_3: {
+          primary: "primary",
+          secondary: "secondary",
+        },
+      };
+    }
+  },
+  selectable: true,
+  slotTemplate: `
+    <template #nested-content="{ row }">
```

Comment:
put all data needed for the table into `row.nestedData`. Including multiple rows for the table.

This makes the example more real life related.

Reply:
why only 1 row in the nested table. add at least 3 and use func for it to reduce the code amount.